cmake/BLTPrivateMacros.cmake

```diff
@@ -463,6 +463,12 @@ macro(blt_split_source_list_by_language)
 
     # Generate source lists based on language
     foreach(_file ${arg_SOURCES})
+        # Assume the user knows what they are doing if using a
```

Comment:
Should we also be printing out any files that cannot be split (i.e. in this list or in the `else()` clause below)?

The assumption that users know what they are doing is probably fine for the users that create these generator expressions, but not necessarily for their consumers.

How about creating a list of any variables that cannot be split and printing them out after the `foreach()`.
We can combine this with a separate variable that indicates if there is a `FATAL_ERROR` condition, which at this point would be a file that is not a generator expression or tht does not have a C- or Fortran extension.

Reply:
we could also filter these before we call split_source_list - however @white238's change is the most surgical fix.

To make other crazy situations easier to debug in the future, it would be good to error if we have an empty extension (and state that fact, showing the input). 
It took me quite a bit to understand where things went off the rails.